### PR TITLE
fix/uint config mismatch

### PR DESCRIPTION
**Pull request by Lord-Kamina**
This was considered in the original implementation of the unsigned configitems. It was later removed in (presumably) a merge error.

I’ve also removed a duplicate function, probably due to the same thing.

### Closes Issue(s)

#838 

### Motivation
QA

### More

- [ ] Added/updated documentation

### CODE REVIEW

File: game/configuration.cc [hunk 1]
```diff
@@ -108,7 +108,11 @@ void ConfigItemXMLLoader::update(ConfigItem& item, xmlpp::Element& elem, unsigne
 			item.setLongDescription(getText(elem, "long"));
 		}
 		else {
-			const auto type = getAttribute(elem, "type");
+			std::string type{getAttribute(elem, "type")};
+			if (item.getType() == "uint" && type == "int") {
+				type = "uint"; // Convert old config values.
```
Comment (Baklap4) [inline game/configuration.cc:113]: Let's double check if this is the only type conversion or if we have some more.
I see we do define: `uint`, `int`,  and `float`

Comment (Lord-Kamina) [reply game/configuration.cc:113]: Yes but I don't think I changed any other types. Checking the history of schema.xml should do the trick either way

**Review by Baklap4 — APPROVED** [review]
LGTM, @taynaron can you try out this fix :)?

**Comment by Lord-Kamina** [comment]
> I'd love to! Can someone please approve and merge https://github.com/performous/performous/pull/837 so I don't have to keep manually applying my changes there to other branches?

Not for now. I'm working on rewriting that whole script, and plan to add better support for it then, but the PR as is (or at least, last I checked earlier today) would break it for everyone else.

If you don't mind me commiting to your branch, I could probably make a temporary stopgap to fix it in line with the current style of the script.

**Comment by Baklap4** [comment]
> I'd love to! Can someone please approve and merge https://github.com/performous/performous/pull/837 so I don't have to keep manually applying my changes there to other branches?

For now it might be easier to make a local branch (which you already have) to contain the changes of #837 and when you work branch of from that branch. Then when you want to create the PR rebase it on master so the commit(s) from #837 aren't there any more.
If you then have to rework the other feature (in this case #840) you locally rebase with #837 (`git rebase -i 837`) and when you're done rebase it again with master `git rebase -i master`. The rebase shouldn't give any conflicts in most of the other code you make. Imho this is easier than constantly cherry-picking the commit and removing it again

**Comment by taynaron** [comment]
I'm all for you committing to my branch if it'll help move things forward :)

**Comment by taynaron** [comment]
I have tested, and the arm version now can run with prefs created on x86_64. I don't have the setup to allow me to test the converse, but I see no reason it wouldn't work.

**Comment by taynaron** [comment]
@Baklap4 Tested, LGTM!

**Pull request merged**